docs/architecture/adr-070-unordered-account.md

```diff
@@ -2,214 +2,137 @@
-un-ordered transaction inclusion.
+transactions and without requiring the use of monotonically increasing sequences. Instead, we propose
+the use of a time-based, ephemeral sequence.
 
 ## Context
 
-As of today, the nonce value (account sequence number) prevents replay-attack and
-ensures the transactions from the same sender are included into blocks and executed
-in sequential order. However it makes it tricky to send many transactions from the
-same sender concurrently in a reliable way. IBC relayer and crypto exchanges are
-typical examples of such use cases.
+Account sequence values serve to prevent replay-attacks and ensure transactions from the same sender are included into blocks and executed
+in sequential order. Unfortunately, this makes it difficult to reliably send many concurrent transactions from the
+same sender. Victims of such limitations include IBC relayers and crypto exchanges.
 
 ## Decision
 
-We propose to add a boolean field `unordered` to transaction body to mark "un-ordered"
-transactions.
+We propose adding a boolean field `unordered` and a uint64 field `timeout_timestamp` to the transaction body.
 
-Un-ordered transactions will bypass the nonce rules and follow the rules described
-below instead, in contrary, the default ordered transactions are not impacted by
-this proposal, they'll follow the nonce rules the same as before.
+Unordered transactions will bypass the traditional account sequence rules and follow the rules described
+below, without impacting traditional ordered transactions; they'll follow the sequence rules the same as before.
 
-When an un-ordered transaction is included into a block, the transaction hash is
-recorded in a dictionary. New transactions are checked against this dictionary for
-duplicates, and to prevent the dictionary grow indefinitely, the transaction must
-specify `timeout_timestamp` for expiration, so it's safe to removed it from the
-dictionary after it's expired.
+We will introduce new storage of time-based, ephemeral unordered sequences using the SDK's existing KV Store library. 
+Specifically, we will leverage the existing x/auth KV store to store the unordered sequences.
 
-The dictionary can be simply implemented as an in-memory golang map, a preliminary
-analysis shows that the memory consumption won't be too big, for example `32M = 32 * 1024 * 1024`
-can support 1024 blocks where each block contains 1024 unordered transactions. For
-safety, we should limit the range of `timeout_timestamp` to prevent very long expiration,
-and limit the size of the dictionary.
+When an unordered transaction is included in a block, a concatenation of the `timeout_timestamp` and sender’s bech32 address
+will be recorded to state (i.e. `542939323/cosmos1v1234567890AbcDeF`). In cases of multi-party signing, we will use a sorted,
+comma-separated list of the public key addresses that signed the transaction (i.e. `5532231/5AEKNF,5AEKNE,5AEKNR`)
```

Comment:
Isn't it simpler to just store one entry per signer? I guess this is okay too

Reply:
updated to 1 entry per signer